src/tools/config_common.h

```diff
@@ -1,6 +1,49 @@
 #include <base/system.h>
 #include <engine/storage.h>
 
+struct ListDirectoryContext
+{
+	const char *pPath;
+	IStorage *pStorage;
+};
+
+void ProcessItem(const char *pItemName, IStorage *pStorage)
+{
+	char aConfig[2048];
+
+	size_t Len = (size_t)str_length(pItemName) + 1; // including '\0'
+	if(Len > sizeof(aConfig))
+	{
+		dbg_msg("config_common", "can't process overlong filename '%s'", pItemName);
+		return;
+	}
+
+	if(!str_endswith(pItemName, ".map"))
+	{
+		dbg_msg("config_common", "can't process non-map file '%s'", pItemName);
+		return;
+	}
+
+	str_copy(aConfig, pItemName, sizeof(aConfig));
+	aConfig[Len - sizeof(".map")] = 0;
+	str_append(aConfig, ".cfg", sizeof(aConfig));
+	dbg_msg("config_common", "processing '%s'", pItemName);
+	Process(pStorage, pItemName, aConfig);
+}
+
+static int ListdirCallback(const char *pItemName, int IsDir, int StorageType, void *pUser)
+{
+	if(!IsDir)
+	{
+		ListDirectoryContext Context = *((ListDirectoryContext *)pUser);
+		char aName[2048];
+		str_format(aName, sizeof(aName), "%s\\%s", Context.pPath, pItemName);
```

Comment:
```suggestion
		str_format(aName, sizeof(aName), "%s/%s", Context.pPath, pItemName);
```

Use slash, backslash only works on Windows.